File: abc/abc319/b.cpp
// https://atcoder.jp/contests/abc319/submissions/45363785

#include <bits/stdc++.h>
using namespace std;

int main() {
  int n;
  cin >> n;

  cout << "1";
  for(int i = 1; i <= n; i++) {
    bool check = false;
    for(int j = 1; j <= 9; j++) {
      if(n % j != 0) continue;
      if(i % (n / j) == 0) {
        cout << j;
        check = true;
        break;
      }
    }

    if(!check) {
      cout << "-";
    }
  }
}

File: abc/abc319/b.py
# https://atcoder.jp/contests/abc319/submissions/45429350

n = int(input())

for i in range(n + 1):
  ans = 0
  for j in range(1, 10):
    if n % j == 0 and i % (n // j) == 0:
      ans = j
      break
  print(ans if ans != 0 else '-', end = '')
